bitcoin/tx.c

```diff
@@ -983,18 +983,14 @@ struct amount_sat change_amount(struct amount_sat excess, u32 feerate_perkw,
 				size_t total_weight)
 {
 	struct amount_sat fee = change_fee(feerate_perkw, total_weight);
-	struct amount_sat dust_limit;
 
 	if (!amount_sat_sub(&excess, excess, fee))
 		return AMOUNT_SAT(0);
 
-	if (chainparams->is_elements)
-		dust_limit = AMOUNT_SAT(330); /* P2WPKH */
-	else
-		dust_limit = AMOUNT_SAT(330); /* P2TR */
-
-	/* Must be non-dust */
-	if (!amount_sat_greater_eq(excess, dust_limit))
+	// Change is P2TR (Bitcoin) or P2WPKH (Elements) both have
+	// dust limit of 330 sat. Legacy types (P2PKH/P2SH) use 546 sat
+	// but we dont create those as change outputs 
+	if (!amount_sat_greater_eq(excess, AMOUNT_SAT(330)))
```

Comment:
The check for elements also got lost somewhere. The logic matches but elements should not create these sub dust outputs since it doesn't use p2tr yet